### DISCUSSION TITLE
Csla.ApplicationContext.User.Identity lost after updating from 4.11.2 to 5.4.2

### DISCUSSION THREAD

**Discussion by michaelcsikos** [MarimerLLC/csla#2298] [Questions] [answered]
Our login system for a .NET 4.8 Windows Forms app is based on `CslaIdentityBase<T>` and `CslaPrincipal`. After successfully logging in, the `Csla.ApplicationContext.User` is immediately reverting back to `UnauthenticatedPrincipal` on the main thread and form. Is there a new config option that I'm missing, or do we have to move everything over to the `ClaimsPrincipal` style?

**Comment by rockfordlhotka** [comment, marked as answer]
Interesting. I wonder if they (Microsoft) changed the behavior of principal management in Windows Forms to match WPF?

WinForms used to be different from WPF, and in WPF the CSLA context manager maintains the User property in a static instead of on the current thread. The WinForms context manager stores the User property on the current thread, which _used to be the correct solution_ but maybe isn't anymore?

**Reply by rockfordlhotka** [reply]
That said, looking at the [WinForms context manager](https://github.com/MarimerLLC/csla/blob/main/Source/Csla.Windows.Shared/ApplicationContextManager.cs), I see that it now does use a static also.

That makes sense, given that a while back I thought I changed the behavior based on a Microsoft change like I mentioned in my previous comment. Apparently my memory isn't entirely gone 😊

So then the question is why your code isn't using the WinForms context manager like it should (and must)? You do have the `Csla.Windows.Forms` package referenced in your project? Maybe you need to add configuration on app startup to force the use of that particular context manager?

**Reply by michaelcsikos** [reply]
Thank you! I had indeed left out the `Csla.Windows.Forms` package. I've been updating projects to use the `Project Sdk` format, and missed this package.
